semanticdb-kotlinc/src/main/kotlin/com/sourcegraph/semanticdb_kotlinc/Analyzer.kt

```diff
@@ -19,24 +27,37 @@ class Analyzer(
 ) : AnalysisHandlerExtension {
     private val globals = GlobalSymbolsCache()
 
+    private val messageCollector =
+        CompilerConfiguration()
+            .get(CLIConfigurationKeys.MESSAGE_COLLECTOR_KEY, MessageCollector.NONE)
+
     override fun analysisCompleted(
         project: Project,
         module: ModuleDescriptor,
         bindingTrace: BindingTrace,
         files: Collection<KtFile>
-    ): AnalysisResult? {
-        val resolver = DescriptorResolver(bindingTrace).also { globals.resolver = it }
-        for (file in files) {
-            val lineMap = LineMap(project, file)
-            val document = SemanticdbVisitor(sourceroot, resolver, file, lineMap, globals).build()
-            semanticdbOutPathForFile(file)?.apply {
-                Files.write(this, TextDocuments { addDocuments(document) }.toByteArray())
+    ): AnalysisResult? =
+        try {
+            val resolver = DescriptorResolver(bindingTrace).also { globals.resolver = it }
+            for (file in files) {
+                try {
+                    val lineMap = LineMap(project, file)
+                    val document =
+                        SemanticdbVisitor(sourceroot, resolver, file, lineMap, globals).build()
+                    semanticdbOutPathForFile(file)?.apply {
+                        Files.write(this, TextDocuments { addDocuments(document) }.toByteArray())
+                    }
+                    callback(document)
+                } catch (e: Exception) {
+                    handleException(e)
+                }
             }
-            callback(document)
-        }
 
-        return super.analysisCompleted(project, module, bindingTrace, files)
-    }
+            super.analysisCompleted(project, module, bindingTrace, files)
+        } catch (e: Exception) {
```

Comment:
Does Kotlin have something similar to Scala's `NonFatal` handler? 

https://sourcegraph.com/github.com/scala/scala/-/blob/src/library/scala/util/control/NonFatal.scala?L42

You typically don't want to catch `Exception` since it swallows fatal exceptions like OOM or VM failures that should propagate.